src/video_core/renderer_opengl/gl_buffer_cache.cpp

```diff
@@ -39,17 +45,18 @@ GLintptr OGLBufferCache::UploadMemory(Tegra::GPUVAddr gpu_addr, std::size_t size
     AlignBuffer(alignment);
     const GLintptr uploaded_offset = buffer_offset;
 
-    Memory::ReadBlock(*cpu_addr, buffer_ptr, size);
+    if (host_ptr) {
```

Comment:
This can be inverted into a guard clause like:

```suggestion
    if (!host_ptr) {
        return uplocaded_offset;
    }
```

letting the memcpy sit next to the related buffer_ptr manipulation down below.